pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -357,16 +368,20 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 				return fmt.Errorf("error creating TaskRun called %s for PipelineTask %s from PipelineRun %s: %s", rprt.TaskRunName, rprt.PipelineTask.Name, pr.Name, err)
 			}
 		}
+		go c.timeoutHandler.WaitPipelineRun(pr)
 	}
 
 	before := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
 	after := resources.GetPipelineConditionStatus(pr.Name, pipelineState, c.Logger, pr.Status.StartTime, pr.Spec.Timeout)
 
+	statusKey := fmt.Sprintf("%s/%s/%s", pipelineRunControllerName, pr.Namespace, pr.Name)
```

Comment:
We may want to use `getPipelineRunKey` (same for `TaskRun`) to make sure we use the same key always :angel: